### PR TITLE
Added location CRUD operations

### PR DESCRIPTION
**Pull request by sofiiagran**
Fixed location crud operations so it works with frontend (see create-event PR branch, if not merged to dev already).

Added unit tests :)) 

image

### CODE REVIEW

File: src/locations/locationModel.ts [hunk 1]
```diff
@@ -6,10 +6,11 @@ export class Location extends BaseModel {
   zip_code: string;
   country: string;
   street_address: string;
-  building_number: string;
-  default_opening_hours: string;
-  default_timeslots: string[];
-  default_number_of_boxes: number;
+  apartment_etc?: string;
+  default_opening_hours?: string;
```
Comment (loui981c) [inline src/locations/locationModel.ts:10]: opening_hours? er det ikke mere relevant for event?

File: src/locations/locationService.ts [hunk 2]
```diff
@@ -97,6 +97,7 @@ export class LocationService extends BaseService<Location> {
   protected validateAndMapRecord(
     records: Location,
   ): Result<Location, BaseError> {
+    records = new Location(records);
```
Comment (loui981c) [inline src/locations/locationService.ts:100]: brug records i return i stedet for at lave en new location 2 gange

File: src/reservations/reservationService.ts [hunk 3]
```diff
@@ -159,6 +159,7 @@ export class ReservationService extends BaseService<Reservation> {
     records: Reservation,
   ): Result<Reservation, BaseError> {
     const validationResult = this.validate(records, true);
+    records = new Reservation(records);
```
Comment (loui981c) [inline src/reservations/reservationService.ts:162]: same som overfor, no need for 2 gange new

**Review by loui981c — CHANGES_REQUESTED** [review]

File: src/locations/locationModel.ts (hunk 1 above)
Comment (sofiiagran) [reply src/locations/locationModel.ts:10]: nej for en location har jo default opening hours, altså på sundholmsvej 28a er det altid default kl 15-16

File: src/locations/locationService.ts (hunk 2 above)
Comment (sofiiagran) [reply src/locations/locationService.ts]: fixed

**Comment by sofiiagran** [comment]
i fixed @loui981c

**Pull request merged**